### PR TITLE
Revoking of rights does not support apps 

### PR DESCRIPTION
**Pull request by andreasisnes**
## Description
* Update Attribute Resolver to handle fetching og user and organization using just partyId 
* Update revoke delegation to support new URNs 

## Related Issue(s)
- #613 

## Developer/Reviewer Checklist
- [ ] **Your** code builds clean without any errors or warnings
- [ ] No changes to config/appsettings or environment variables created in separate linked PR
- [ ] Manual testing done (required)
- [ ] Relevant integration test added (required for functional changes)
- [ ] Relevant automated test added (required for functional changes)
- [ ] All integration tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.AccessManagement.Core/Helpers/DelegationHelper.cs [hunk 1]
```diff
@@ -488,6 +490,30 @@ public static List<RequestToDelete> GetRequestToDeleteResourceRegistryService(in
             };
         }
 
+        /// <summary>
+        /// Builds a RequestToDelete request model for revoking all delegated rules for a resource registry service
+        /// </summary>
+        public static List<RequestToDelete> GetRequestToDeleteResourceRegistryService(int authenticatedUserId, IEnumerable<AttributeMatch> resource, int fromPartyId, IEnumerable<AttributeMatch> to)
```
Comment (jonkjetiloye) [inline src/Altinn.AccessManagement.Core/Helpers/DelegationHelper.cs:496]: Name is misleading as it's not only `ResourceRegistryService`.

File: test/Bruno/Altinn.AccessManagement/Manual Test Collection/RightsInternal/GetOfferedRevoke/Kasper Revoke Offered from Rimstad.bru [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+headers {
+  Authorization: {{apiKey}}
+}
+
+body:json {
+  
+    {
+      "to": [
+        {
+          "id": "urn:altinn:person:partyid",
```
Comment (jonkjetiloye) [inline test/Bruno/Altinn.AccessManagement/Manual Test Collection/RightsInternal/GetOfferedRevoke/Kasper Revoke Offered from Rimstad.bru:22]: If this is to be a valid test case for the RightsInternal API Offered Revoke as used from Altinn 2, this should use
"to": "urn:altinn:userid"

Comment (ivartryti) [reply test/Bruno/Altinn.AccessManagement/Manual Test Collection/RightsInternal/GetOfferedRevoke/Kasper Revoke Offered from Rimstad.bru]: fixed test

File: src/Altinn.AccessManagement.Core/Helpers/DelegationHelper.cs (hunk 1 above)
Comment (ivartryti) [reply src/Altinn.AccessManagement.Core/Helpers/DelegationHelper.cs]: Renamed "GetRequestToDeleteResourceRegistryService" to "GetRequestToDeleteResource"

File: test/Bruno/Altinn.AccessManagement/Manual Test Collection/RightsInternal/GetOfferedRevoke/Kasper Revoke Offered from Rimstad.bru (hunk 2 above)
Comment (ivartryti) [reply test/Bruno/Altinn.AccessManagement/Manual Test Collection/RightsInternal/GetOfferedRevoke/Kasper Revoke Offered from Rimstad.bru]: Didn't actually fix test! Reopening this one.

Comment (ivartryti) [reply test/Bruno/Altinn.AccessManagement/Manual Test Collection/RightsInternal/GetOfferedRevoke/Kasper Revoke Offered from Rimstad.bru]: Added old "UserAttribute" to resolver and GetRequestToDeleteResource, and the test is fixed now.

**Comment by jonkjetiloye** [comment]
Need to fix the style-cop build errors which breaks the `Code test and analysis`

File: src/Altinn.AccessManagement.Core/Asserters/AttributeMatchAsserter.cs [hunk 3]
```diff
@@ -121,11 +127,16 @@ public static void IsDelegatableResource(this IAssert<AttributeMatch> assert, ID
...
                     assert.HasAttributeTypes(Urn.Altinn.Person.UserId),
                     assert.HasAttributeTypes(Urn.Altinn.Person.PartyId),
                     assert.HasAttributeTypes(Urn.Altinn.Organization.IdentifierNo),
+                    assert.HasAttributeTypes(Urn.Altinn.Organization.Uuid),
                     assert.HasAttributeTypes(Urn.Altinn.EnterpriseUser.Username),
-                    assert.HasAttributeTypes(Urn.Altinn.Organization.PartyId)),
+                    assert.HasAttributeTypes(Urn.Altinn.EnterpriseUser.Uuid),
+                    assert.HasAttributeTypes(Urn.Altinn.Organization.PartyId),
+                    assert.HasAttributeTypes(AltinnXacmlConstants.MatchAttributeIdentifiers.PartyAttribute),
+                assert.HasAttributeTypes(AltinnXacmlConstants.MatchAttributeIdentifiers.UserAttribute)),
```
Comment (jonkjetiloye) [inline src/Altinn.AccessManagement.Core/Asserters/AttributeMatchAsserter.cs:139]: Need to fix the style-cop build errors which breaks the Code test and analysis

**Review by jonkjetiloye — APPROVED** [review]
once the build issue is fixed this looks good to go! 👍

**Pull request merged**